src/switchroot/ostree-prepare-root.c

```diff
@@ -259,13 +259,27 @@ main (int argc, char *argv[])
 {
   char srcpath[PATH_MAX];
   struct stat stbuf;
+  gboolean booted = FALSE;
   g_autoptr (GError) error = NULL;
 
   if (argc < 2)
-    err (EXIT_FAILURE, "usage: ostree-prepare-root SYSROOT");
+    err (EXIT_FAILURE, "usage: ostree-prepare-root SYSROOT [KERNEL_CMDLINE]");
   const char *root_arg = argv[1];
 
-  g_autofree char *kernel_cmdline = read_proc_cmdline ();
+  /* Check if we're in initramfs or not */
+  if (fstatat (AT_FDCWD, OTCORE_RUN_BOOTED, &stbuf, 0) == 0)
+    booted = (stbuf.st_mode & S_IFMT) == S_IFREG;
+
+  g_autofree char *kernel_cmdline;
+  if (argc < 3)
+    {
+      kernel_cmdline = read_proc_cmdline ();
+    }
+  else
+    {
+      kernel_cmdline = argv[2];
```

Comment:
Awesome! This looks like a really nice change and should crucially really help us for testing ostree-prepare-root outside of the initramfs in general (should be totally doable to run in a container)

Reply:
Simple and effective :)